dnsaas/admin.py

```diff
@@ -0,0 +1,427 @@
+from django_extensions.admin import ForeignKeyAutocompleteAdmin
+from django.contrib import admin
+from django.contrib.admin.sites import AdminSite
+from django.forms import (
+    HiddenInput,
+    NullBooleanSelect,
+    ModelForm,
+    ValidationError,
+    ModelChoiceField,
+)
+from powerdns.models.templates import (
+    DomainTemplate,
+    RecordTemplate,
+)
+
+from powerdns.models.powerdns import (
+    CryptoKey,
+    Domain,
+    DomainMetadata,
+    Record,
+    SuperMaster,
+)
+from powerdns.models.tsigkeys import TsigKey
+from powerdns.utils import Owned, DomainForRecordValidator, is_owner
+
+
+admin_site2 = AdminSite('admin2')
+
+
+#TODO: is this really needed?
+class CopyingAdmin(admin.ModelAdmin):
+
+    field_prefix = ''
+    target_prefix = ''
+
+    def get_form(self, request, obj=None, **kwargs):
+        form = super().get_form(request, obj, **kwargs)
+        from_pk = request.GET.get(self.from_field)
+        if from_pk is not None:
+            self.from_object = self.FromModel.objects.get(pk=from_pk)
+            for field in self.CopyFieldsModel.copy_fields:
+                form.base_fields[field[len(self.field_prefix):]].initial = \
+                    getattr(self.from_object, field[len(self.target_prefix):])
+        else:
+            self.from_object = None
+        return form
+
+
+from rules.contrib.admin import ObjectPermissionsModelAdmin
+class OwnedAdmin(ForeignKeyAutocompleteAdmin, ObjectPermissionsModelAdmin):
+    """Admin for models with owner field"""
+
+    def save_model(self, request, object_, form, change):
+        if object_.owner is None:
+            object_.owner = request.user
+        #TODO: rm it?
+        #object_.email_owner(request.user)
+        super(OwnedAdmin, self).save_model(request, object_, form, change)
+
+    #def get_related_filter(self, model, request):
+    #    #TODO:: i assume this fn is up to remove
+    #    return super(OwnedAdmin, self).get_related_filter(model, request)
+    #    user = request.user
+    #    if not issubclass(model, Owned) or rules.is_superuser(user):
+    #        return super(OwnedAdmin, self).get_related_filter(model, request)
+    #    return models.Q(owner=user)
+
+
+
+
+from django.utils.translation import ugettext_lazy as _
+from django.db import models
+try:
+    from django.contrib.admin import SimpleListFilter
```

Comment:
do we need this try-except? When this exception will occur? (django version related?)